Lib/test/test_ntpath.py

```diff
@@ -133,16 +133,23 @@ def test_splitdrive(self):
         tester('ntpath.splitdrive("\\\\?\\c:")', ("\\\\?\\c:", ""))
         tester('ntpath.splitdrive("\\\\?\\c:\\")', ("\\\\?\\c:", "\\"))
         tester('ntpath.splitdrive("\\\\?\\c:\\dir")', ("\\\\?\\c:", "\\dir"))
-        tester('ntpath.splitdrive("\\\\?\\UNC")', ("", "\\\\?\\UNC"))
-        tester('ntpath.splitdrive("\\\\?\\UNC\\")', ("", "\\\\?\\UNC\\"))
-        tester('ntpath.splitdrive("\\\\?\\UNC\\server\\")', ("\\\\?\\UNC\\server\\", ""))
+        tester('ntpath.splitdrive("\\\\?\\UNC")', ("\\\\?\\UNC", ""))
+        tester('ntpath.splitdrive("\\\\?\\UNC\\")', ("\\\\?\\UNC\\", ""))
+        tester('ntpath.splitdrive("\\\\?\\UNC\\server\\")', ("\\\\?\\UNC\\server", "\\"))
         tester('ntpath.splitdrive("\\\\?\\UNC\\server\\share")', ("\\\\?\\UNC\\server\\share", ""))
         tester('ntpath.splitdrive("\\\\?\\UNC\\server\\share\\dir")',
                ("\\\\?\\UNC\\server\\share", "\\dir"))
         tester('ntpath.splitdrive("\\\\?\\VOLUME{00000000-0000-0000-0000-000000000000}\\spam")',
                ('\\\\?\\VOLUME{00000000-0000-0000-0000-000000000000}', '\\spam'))
         tester('ntpath.splitdrive("\\\\?\\BootPartition\\")', ("\\\\?\\BootPartition", "\\"))
 
+        # gh-96290: support partial/invalid UNC drives
+        tester('ntpath.splitdrive("//")', ("//", ""))  # empty server & missing share
+        tester('ntpath.splitdrive("///")', ("//", "/"))  # empty server & empty share
+        tester('ntpath.splitdrive("///y")', ("///y", ""))  # empty server & valid share
+        tester('ntpath.splitdrive("//x")', ("//x", ""))  # valid server & missing share
+        tester('ntpath.splitdrive("//x/")', ("//x", "/"))  # valid server & empty share
```

Comment:
Maybe change the wording in the comments to "non-empty" instead of "valid".

```suggestion
        tester('ntpath.splitdrive("///y")', ("///y", ""))  # empty server & non-empty share
        tester('ntpath.splitdrive("//x")', ("//x", ""))  # non-empty server & missing share
        tester('ntpath.splitdrive("//x/")', ("//x", "/"))  # non-empty server & empty share
```